test/UriTest.php

```diff
@@ -136,6 +136,13 @@ public function testWithHostReturnsSameInstanceWithProvidedHostIsSameAsBefore()
         $this->assertSame('https://user:pass@local.example.com:3001/foo?bar=baz#quz', (string) $new);
     }
 
+    public function testWithHostWithGermanUmlaut()
+    {
+        $uri = new Uri('https://körberl.taugl.online');
+        $new = $uri->withHost('körberl.taugl.online');
+        $this->assertSame($uri, $new);
```

Comment:
Looks like wrong test because you put same value and `Url` is still same instance, so nothing changed. I suggest to change first host to something different and check result of `getHost()` like in `testUriWithGermanUmlaut`

Reply:
Absolutely... I just fixed it and added mb_strtolower to method `withHost` - this time it's not working without it!